src/Auth/OAuth2Protocol.js

```diff
@@ -1,25 +1,132 @@
+   * @type {string|null}
+   */
+  username = null;
+
+  /**
+   * The password for authentication.
+   * @type {string|null}
+   */
+  password = null;
+
+  /**
+   * The client identifier.
+   * @type {string|null}
+   */
+  clientId = null;
+
+  /**
+   * The requested scope of access.
+   * @type {string|null}
+   */
+  scope = null;
+
+  /**
+   * The state parameter for CSRF protection.
+   * @type {string|null}
+   */
+  state = null;
+
+  /**
+   * The login hint for authentication.
+   * @type {string|null}
+   */
+  login_hint = null;
+
+  /**
+   * The PKCE code challenge method.
+   * @type {string|null}
+   */
+  codeChallengeMethod = null;
+
+  /**
+   * The current authentication token.
+   * @typedef {Object|null} token
+   * @property {string} access_token
+   * @property {string} refresh_token
+   */
+  token=null;
+
+  /**
+   * Flag indicating if a token refresh is currently in progress.
+   * @type {boolean}
+   */
+  tokenRefreshInProgress = false;
+
+  /**
+   * Callback function for token updates.
+   * @type {Function|null}
+   */
+  onAuthorizationInfoUpdate = null;
+
+  /**
+   * Creates an instance of OAuth2Protocol.
+   * Initializes error messages and default property values.
+   */
   constructor() {
     this.fieldMissingMessage = 'Field "{0}" required for "OAuth 2.0" authentication is missing!';
     this.unsupportedGrantTypeMessage = 'Grant type "{0}" not supported. Please use either "code" type or "password" type.';
     this.unsupportedCodeChallengeMethodMessage = 'Code challenge (PKCE) method "{0}" not supported. Please use "S256" method or disable it.';
-
-    this.authEndpoint = null;
-    this.tokenEndpoint = null;
-    this.grantType = null;
-    this.username = null;
-    this.password = null;
-    this.clientId = null;
-    this.scope = null;
-    this.state = null;
-    this.login_hint=null;
-    this.codeChallengeMethod = null;
-    this.token = null;
-    this.tokenRefreshInProgress=false;
-
-    this.onAuthorizationInfoUpdate = null;
   }
 
+  /**
+   * Initializes the OAuth2 protocol with the provided configuration.
+   *
+   * @param {Object} config - Configuration object containing OAuth2 parameters
+   * @param {string} config.token_endpoint - The token endpoint URL
+   * @param {string} config.grant_type - The grant type (password, refresh_token, etc.)
+   * @param {string} config.client_id - The client ID
+   * @param {string} [config.scope] - Optional scope
+   * @param {string} [config.state] - Optional state
+   * @param {string} [config.code_challenge_method] - Optional PKCE code challenge method
+   * @param {string} [config.username] - Username for password grant type
+   * @param {string} [config.password] - Password for password grant type
+   * @param {string} [config.login_hint] - Login hint for password grant type
+   * @returns {Promise<void>}
+   * @throws {Error} If required configuration values are missing or grant type is unsupported
+   */
   async init(config) {
```

Comment:
why async ?. Having init method it is a bad practice => implement proper constructor